pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -639,9 +676,30 @@ func (lb *LoadBalancer) CreateTCPLoadBalancer(name, region string, externalIP ne
 	}
 
 	status := &api.LoadBalancerStatus{}
-	status.Ingress = []api.LoadBalancerIngress{{IP: vip.Address}}
+
+	if lb.opts.FloatingNetworkId == "" {
```

Comment:
(style) I'd invert these, to avoid the branches diverging further over time:

```go
status.Ingress = []api.LoadBalancerIngress{{IP:vip.Address}}

if lb.opts.FloatingNetworkId != "" {
    // floating ip logic below
    // floatIP := ...
    status.Ingress = append(status.Ingress, {IP: floatIP.FloatingIP})
}

return status, nil